models/issues/issue_update.go

```diff
@@ -696,85 +696,100 @@ func UpdateReactionsMigrationsByType(gitServiceType api.GitServiceType, original
 
 // DeleteIssuesByRepoID deletes issues by repositories id
 func DeleteIssuesByRepoID(ctx context.Context, repoID int64) (attachmentPaths []string, err error) {
-	deleteCond := builder.Select("id").From("issue").Where(builder.Eq{"issue.repo_id": repoID})
-
+	// MariaDB has a performance bug: https://jira.mariadb.org/browse/MDEV-16289
+	// so here it uses "DELETE ... WHERE IN" with pre-queried IDs.
 	sess := db.GetEngine(ctx)
-	// Delete content histories
-	if _, err = sess.In("issue_id", deleteCond).
-		Delete(&ContentHistory{}); err != nil {
-		return nil, err
-	}
 
-	// Delete comments and attachments
-	if _, err = sess.In("issue_id", deleteCond).
-		Delete(&Comment{}); err != nil {
-		return nil, err
-	}
+	for {
+		issueIDs := make([]int64, 0, db.DefaultMaxInSize)
 
-	// Dependencies for issues in this repository
-	if _, err = sess.In("issue_id", deleteCond).
-		Delete(&IssueDependency{}); err != nil {
-		return nil, err
-	}
+		err := sess.Table(&Issue{}).Where("repo_id = ?", repoID).Limit(db.DefaultMaxInSize).Cols("id").Find(&issueIDs)
```

Comment:
We need an explicit `OrderBy` here.

Reply:
Why?

Reply:
To keep the query order explicitly and not affected by database global setting.

Reply:
Ah, I overlooked the endless `for` loop.
That one wouldn't be needed either anymore with my approach.